### ISSUE TITLE
Umlauts are not displayed

### ISSUE THREAD

**Issue by RC2225** [luciusDXL/TheForceEngine#210]
The German umlauts are not working in the messages displayed in game. The text should be Höhenüberprüfung

![umlaute](https://user-images.githubusercontent.com/5329274/215901586-2348585b-4f4f-4440-85e0-117d21b6186e.png)

**Comment by mlauss2** [comment]
I can confirm, working on it.

**Comment by mlauss2** [comment]
It's a weird signedness issue; all glyphs with index > 127 are not printed.
This patch fixes it for me:
```diff
diff --git a/TheForceEngine/TFE_DarkForces/hud.cpp b/TheForceEngine/TFE_DarkForces/hud.cpp
index a124bb59..ae40c4c7 100644
--- a/TheForceEngine/TFE_DarkForces/hud.cpp
+++ b/TheForceEngine/TFE_DarkForces/hud.cpp
@@ -1076,7 +1076,7 @@ namespace TFE_DarkForces
                        fixed16_16 x0 = xf;
 
                        fixed16_16 fWidth = mul16(intToFixed16(font->width), xScale);
-                       for (char c = *msg; c != 0;)
+                       for (unsigned char c = *(unsigned char *)msg; c != 0;)
                        {
                                if (c == '\n')
                                {
@@ -1095,7 +1095,7 @@ namespace TFE_DarkForces
                                        xf += mul16(intToFixed16(font->horzSpacing + glyph->width), xScale);
                                }
                                msg++;
-                               c = *msg;
+                               c = *(unsigned char *)msg;
                        }
                }
        }


```

**Comment by TP555** [comment]
Hi

@mlauss2

Ah nice Work so far  :+1: 

You did fix it with the Real ÄÖÜ , or without German umlauts  ?

Ä =AE
Ö=OE
U=UE

I did fix this without it for a while , by Edit the Text.msg Extracted from the Dark.Gob File
https://the-force-engine.freeforums.net/thread/162/german-words-issue

**Comment by mlauss2** [comment]
It's a code fix, I'll do a pull request or if Lucius is reading this, he can probably do this 2 line change himself.

So siehts bei mir jetzt aus:
![image](https://user-images.githubusercontent.com/123231678/216154465-f3e22742-0406-4949-b760-1dfa2ad151d9.png)

**Comment by TP555** [comment]
@mlauss2 

That all fine , good Work   :+1:

**Comment by mlauss2** [comment]
> your Fix can now show Unicodes ? see Post #5 !

No, the code only does ASCII, and the Font used to render these texts only implements glyphs for codes 33-150.
For real unicode support, some larger code surgery is necessary.

**Comment by luciusDXL** [comment]
I merged the PR from mlauss, but then went through the code and converted the message and API to use u8* strings. This also fixed the 320x200 case which would still have been an issue with the PR.
